lib/process-data-availability.js

```diff
@@ -212,22 +197,33 @@ const getAvailabilityAggregatesByResourceTemplate = () => {
  * @param {Object} availabilityReport the RESO availability report JSON to process.
  * @returns a JSON availability report with the appropriate roll-ups and aggregates.
  */
-const processDataAvailability = async (availabilityReport) => {
+const processDataAvailability = async availabilityReport => {
   //iterate over each field and lookup and compute their availabilities
   const { resources, fields, lookups, lookupValues, ...rest } = availabilityReport;
 
   const transformed = getAvailabilityTemplate();
   const standardFieldCache = createStandardFieldCache(standardMetadata.fields);
 
   const resourceCounts = {};
-  resources.forEach(
-    (resource) =>
-      (resourceCounts[resource.resourceName] = resource.numRecordsFetched)
-  );
+  const nestedResourcesCount = {};
```

Comment:
This could be a `reduce` operation that you destrucure both results from. 

```
const { resourceCounts = {}, expandedResourceCounts = {} } = resources.reduce(() => {
  /* TODO */ 
} , { 
  resourceCounts: {}, 
  expandedResourceCounts: {} 
});
```